splink/input_column.py

```diff
@@ -118,135 +159,171 @@ def parse_input_name_to_sqlglot_tree(self) -> Expression:
+                )
+
+        if tree.find(exp.Bracket):
+            # Pop the column, leaving the bracket as the remaining section of the tree
+            return ColumnTreeBuilder.from_sqlglot_column(
+                column_tree=tree.find(exp.Column),
+                column_reference=tree.find(exp.Literal),
             )
 
-        tree_signature = sqlglot_tree_signature(tree)
-        valid_signatures = ["column identifier", "bracket column literal identifier"]
-
-        if tree_signature in valid_signatures:
-            return tree
+        # If the column has already been quoted, parse it
+        if tree.find(exp.Identifier).args.get("quoted"):
+            return ColumnTreeBuilder.from_sqlglot_column(column_tree=tree)
+
+        # If our column does not contain a quoted identifier, we can safely return
+        # the column builder
+        return ColumnTreeBuilder(name)
+
+    @staticmethod
+    def manually_split_identifier_and_index(identifier: str) -> [str, str]:
+        # Manually pull out the bracket index for an unparseable column.
+        # This ensures we correctly parse cases such as: sur name['lat']
+        if identifier.endswith("]") and "[" in identifier:
+            split_index = identifier.rfind("[")
+            before_bracket = identifier[:split_index].strip()
+            after_bracket = identifier[split_index:].strip("[]")
+            return before_bracket, after_bracket
         else:
-            # e.g. SUR name parses to 'alias column identifier identifier'
-            # but we want "SUR name"
-            tree = sqlglot.parse_one(
-                f"{q_s}{self.input_name}{q_e}", read=self._sql_dialect
-            )
-            return tree
-
-    def from_settings_obj_else_default(self, key, schema_key=None):
-        # Covers the case where no settings obj is set on the comparison level
-        if self._settings_obj:
-            return getattr(self._settings_obj, key)
-        else:
-            if not schema_key:
-                schema_key = key
-            return default_value_from_schema(schema_key, "root")
+            return identifier, None
 
     @property
-    def gamma_prefix(self) -> str:
-        return self.from_settings_obj_else_default(
-            "_gamma_prefix", "comparison_vector_value_column_prefix"
-        )
-
-    @property
-    def bf_prefix(self) -> str:
+    def _bf_prefix(self):
         return self.from_settings_obj_else_default(
             "_bf_prefix", "bayes_factor_column_prefix"
         )
 
     @property
-    def tf_prefix(self) -> str:
+    def _tf_prefix(self):
         return self.from_settings_obj_else_default(
             "_tf_prefix", "term_frequency_adjustment_column_prefix"
         )
 
+    def _copy_with_new_column_builder(self, new_builder) -> InputColumn:
+        input_column_copy = deepcopy(self)
+        input_column_copy.column_tree_builder = new_builder
+        return input_column_copy
+
+    def change_input_column_name(self, new_name: str):
```

Comment:
Not at present and I think it might make sense to remove it for this PR.

The logic was that in the future it's likely that we would want to:
* Create a column with various manipulations queued up
* Create a second copy with identical manipulations, with a different name.